pkg/operator/sync.go

```diff
@@ -998,6 +998,18 @@ func (optr *Operator) syncMachineOSBuilder(config *renderConfig) error {
 // Determines if the Machine OS Builder deployment is in the correct state
 // based upon whether we have opted-in pools or not.
 func (optr *Operator) reconcileMachineOSBuilder(mob *appsv1.Deployment) error {
+	// Access current feature gates
+	fg, err := optr.fgAccessor.CurrentFeatureGates()
+	if err != nil {
+		return fmt.Errorf("could not get feature gates: %w", err)
+	}
+
+	// Check if OnClusterBuild feature gate is enabled
+	if !fg.Enabled(configv1.FeatureGateOnClusterBuild) {
```

Comment:
you might need add a `fg != nil` check here too, I think that can happen sometimes?